returnn/flow.py

```diff
@@ -0,0 +1,129 @@
+from sisyphus import tk
+from sisyphus.delayed_ops import DelayedJoin
+from typing import Optional, List, Union
+
+from i6_core import rasr
+from i6_core.returnn.training import Checkpoint
+
+
+def make_precomputed_hybrid_tf_feature_flow(
+    tf_graph: tk.Path,
+    tf_checkpoint: Checkpoint,
+    output_type: str = "log-posteriors",
+    feature_tensor_name: str = "data",
+    output_tensor_name: str = "output",
+    native_ops: Optional[Union[tk.Path, List[tk.Path]]] = None,
+    tf_fwd_input_name: str = "tf-fwd-input",
+) -> rasr.FlowNetwork:
+    """
+    Create the feature flow for a simple TF network that predicts frame-wise outputs, to be used
+    in combination with the `nn-precomputed-hybrid` feature-scorer setting in RASR.
+
+    The resulting flow is a trivial:
+
+        <link from="<tf_fwd_input_name>" to="tf-fwd:input"/>
+        <node name="tf-fwd" id="$(id)" filter="tensorflow-forward"/>
+        <link from="tf-fwd:log-posteriors" to="network:features"/>
+
+    With the config settings:
+
+        [flf-lattice-tool.network.recognizer.feature-extraction.tf-fwd.input-map.info-0]
+        param-name             = input
+        seq-length-tensor-name = extern_data/placeholders/<feature_tensor_name>/<feature_tensor_name>_dim0_size
+        tensor-name            = extern_data/placeholders/<feature_tensor_name>/<feature_tensor_name>
+
+        [flf-lattice-tool.network.recognizer.feature-extraction.tf-fwd.loader]
+        meta-graph-file    = <tf_graph>
+        required-libraries = <native_ops>
+        saved-model-file   = <tf_checkpoint>
+        type               = meta
+
+        [flf-lattice-tool.network.recognizer.feature-extraction.tf-fwd.output-map.info-0]
+        param-name  = <output_type>
+        tensor-name = <output_tensor_name>/output_batch_major
+
+
+    :param tf_graph: usually the output of a CompileTFGraphJob
+    :param tf_checkpoint: the checkpoint to load the model from, e.g. a ReturnnTrainingJob or similar
+    :param output_type: type of the returned values, see RASR documentation for possible types
+    :param feature_tensor_name: name of the extern data entry to feed the features to
+    :param output_tensor_name: the name of the output tensor, it is expected that "<name>/output_batch_major" exists
```

Comment:
I would either:

- Keep the `output_tensor_name` param, but do not automatically add the `/output_batch_major`, but let the user add this explicitly. Then it is more generic. Further it should be documented here that RASR expects the shape [B,T,D].
- Or, rename the param, maybe to `output_layer_name`, to make it more clear that this is not the tensor name.

Reply:
I would prefer the second approach (tensor is indeed a wrong name), as this function is not intended to be generic but very specific about the `precomputed-hybrid` module of RASR.
@christophmluscher what is your opinion on this?

Reply:
I, too, prefer the second approach.